src/hotspot/cpu/ppc/frame_ppc.inline.hpp

```diff
@@ -68,28 +75,18 @@ inline void frame::setup() {
 
 // Constructors
 
-// Initialize all fields, _unextended_sp will be adjusted in find_codeblob_and_set_pc_and_deopt_state.
+// Initialize all fields
 inline frame::frame() : _sp(nullptr), _pc(nullptr), _cb(nullptr), _oop_map(nullptr), _deopt_state(unknown),
                         _on_heap(false), DEBUG_ONLY(_frame_index(-1) COMMA) _unextended_sp(nullptr), _fp(nullptr) {}
 
-inline frame::frame(intptr_t* sp)
-  : _sp(sp), _pc((address)own_abi()->lr), _cb(nullptr), _oop_map(nullptr),
-    _on_heap(false), DEBUG_ONLY(_frame_index(-1) COMMA) _unextended_sp(sp), _fp(nullptr) {
-  setup();
-}
-
-inline frame::frame(intptr_t* sp, address pc)
-  : _sp(sp), _pc(pc), _cb(nullptr), _oop_map(nullptr),
-    _on_heap(false), DEBUG_ONLY(_frame_index(-1) COMMA) _unextended_sp(sp), _fp(nullptr) {
-  setup();
-}
-
 inline frame::frame(intptr_t* sp, address pc, intptr_t* unextended_sp, intptr_t* fp, CodeBlob* cb)
-  : _sp(sp), _pc(pc), _cb(nullptr), _oop_map(nullptr),
+  : _sp(sp), _pc(pc), _cb(cb), _oop_map(nullptr),
     _on_heap(false), DEBUG_ONLY(_frame_index(-1) COMMA) _unextended_sp(unextended_sp), _fp(fp) {
   setup();
 }
 
+inline frame::frame(intptr_t* sp) : frame(sp, nullptr) {}
+
```

Comment:
Hi, where is this used? 
After you added default arguments to the other constructor, why not add one for pc, too?

Reply:
`explicit frame(intptr_t* sp)` is defined in shared code (frame.hpp). I don't want to add a default argument there.

Reply:
But your question was good. I noticed that we should also simplify the constructor for "pns". See 3rd commit.

Reply:
OK, that makes sense. Thanks